### ISSUE TITLE
Support "Arbitrary module namespace identifier names"

### ISSUE THREAD

**Issue by sosukesuzuki** [import-js/eslint-plugin-import#2355]
Some rules need to be changed for https://github.com/tc39/ecma262/pull/2154?

**related:**

- https://github.com/estree/estree/pull/269
- https://github.com/acornjs/acorn/pull/1090

**Comment by ljharb** [comment]
If so, a PR with test cases would be most helpful.

**Comment by sosukesuzuki** [comment]
> However, I'd expect no changes are needed, since we rely on the parser to handle this.

For example, [no-named-default](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-default.md) should warn for the below code:

```js
import { "default" as bar } from "./bar";
```

However, current one doesn't warn. We should also handle `specifier.imported.value` because `specifier.imported` can be a `Literal` on https://github.com/import-js/eslint-plugin-import/blob/9e4c9a9a06020ed272e49a278c391fcf267e66af/src/rules/no-named-default.js#L20.

These small changes need to be made for many rules.

**Comment by ljharb** [comment]
Good catch! A PR would be appreciated.
